src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TypeEnter.java

```diff
@@ -979,11 +979,13 @@ private void fillPermits(JCClassDecl tree, Env<AttrContext> baseEnv) {
             if (sym.isPermittedExplicit) {
                 ListBuffer<Symbol> permittedSubtypeSymbols = new ListBuffer<>();
                 List<JCExpression> permittedTrees = tree.permitting;
+                baseEnv.info.isPermitsClause = true;
```

Comment:
usually we save the previous value of the field before setting a new value, then inside a `try` we invoke the code that should be aware of the field and then in the `finally` we restore the previous value of the field, see for example how we do for field `AttrContext::ctorPrologue`